packages/shared/src/utils/append-utm-params-to-link.ts

```diff
@@ -0,0 +1,14 @@
+import { UTM_SOURCE } from '@transferto/dapp/src/const';
+import type { UtmParams } from '@transferto/dapp/src/types';
+
+export function appendUTMParametersToLink(link: string, utm: UtmParams) {
+  const searchParams = new URLSearchParams(
+    Object.entries({ ...{ utm_source: UTM_SOURCE }, ...utm }).map(
+      ([key, value]) => [key, String(value)],
+    ),
+  );
+  const updatedLink = link.charAt(link.length - 1) === '/' ? link : `${link}/`;
+  const updatedUrl = `${updatedLink}?${searchParams.toString()}`;
+
+  return updatedUrl;
+}
```

Comment:
Looks much nicer. Small thing:

`Object.entries({ ...{ utm_source: UTM_SOURCE }, ...utm })` -> `Object.entries({ utm_source: UTM_SOURCE, ...utm })`

Reply:
Nice catch! 
Thank you for that recommendation to use native URLSearchParams